src/hooks/useFetchTeamOverview.tsx

```diff
@@ -0,0 +1,43 @@
+import {useEffect, useReducer} from 'react';
+import {teamsOverviewReducer} from 'reducers/TeamsOverviewReducer';
+import {TeamOverviewData, TeamOverviewState} from 'types';
+import {getTeamOverview, getUserData} from '../api';
+
+const useFetchTeamOverview = (teamId: string) => {
+    const initialState: TeamOverviewState = {
+        teamOverviewData: {},
+        loading: true,
+        error: false,
+    };
+
+    const [state, dispatch] = useReducer(teamsOverviewReducer, initialState);
```

Comment:
custom hooks are useful for reusing the logic in different parts of the app. So we can separate the responsibilities and keep the components more readable and cleaner